### PR TITLE
Feature/mtl

### PR DESCRIPTION
**Pull request by Gacardonac**
mtl branch is ready to be merged into the main branch

### CODE REVIEW

File: mtl/mtl.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            self.low = kwargs['low']
+            self.high = kwargs['high']
+            self.left = kwargs['left']
+            self.right = kwargs['right']
+
+        self.__string = None
+        self.__hash = None
+
+    
+    # def negate(self):
```
Comment (wasserfeder) [inline mtl/mtl.py:76]: Adapt the negate function.

File: mtl/mtl.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    #     '''Computes the negation of the MTL formula by propagating the negation
+    #     towards predicates.
+    #     '''
+    #     if self.op == Operation.BOOL:
+    #         self.value = not self.value
+    #     elif self.op == Operation.PRED:
```
Comment (wasserfeder) [inline mtl/mtl.py:82]: Here you would need to create a new MTLFormula object corresponding to a negation operation, and add the predicate as a child. The negation for all other operations remain the same.

File: mtl/mtl.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    #     elif self.op == Operation.NOT:
+    #         return self.child
+    #     elif self.op == Operation.UNTIL:
+    #         raise NotImplementedError
+    #     elif self.op in (Operation.ALWAYS, Operation.EVENT):
+    #         self.child = self.child.negate()
+    #     self.op = Operation.negop[self.op]
+    #     return self
+
+    # def pnf(self):
```
Comment (wasserfeder) [inline mtl/mtl.py:97]: The function should work for the most part. You will need to adapt the code for predicates only.

File: mtl/mtl2milp.py [hunk 4]
```diff
@@ -72,37 +54,25 @@ def add_formula_variable(self, formula, t, vtype=grb.GRB.BINARY):
...
             self.variables[state] = dict()
         if t not in self.variables[state]:
-            low, high = self.ranges[state]
-            vtype = self.vtypes[state]
             name='{}_{}'.format(state, t)
-            v = self.model.addVar(vtype=vtype, lb=low, ub=high, name=name)
-            self.variables[state][t] = v
-            print 'Added state:', state, 'time:', t
+            self.variables[state][t] = z
+            print ('Added state:', state, 'time:', t)
```
Comment (wasserfeder) [inline mtl/mtl2milp.py:68]: Delete `print` statements.

**Review by wasserfeder — COMMENTED** [review]
Initial review.

**Comment by Gacardonac** [comment]
The negation function was added, and print statements were deleted.

**Comment by wasserfeder** [comment]
In this branch STL is deleted. I am closing this PR.

**Pull request closed without merging**